apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -348,7 +348,14 @@ internal SubscriberClientImpl(SubscriptionName subscriptionName, IEnumerable<Sub
             GaxPreconditions.CheckNotNull(settings, nameof(settings));
             settings.Validate();
             _modifyDeadlineSeconds = (int)((settings.StreamAckDeadline ?? DefaultStreamAckDeadline).TotalSeconds);
+            // Enforce server-side constraint on _modifyDeadlineSeconds.
+            GaxPreconditions.CheckArgument(_modifyDeadlineSeconds >= 10 && _modifyDeadlineSeconds <= 600, nameof(settings.StreamAckDeadline), "Must be between 10 and 600 seconds");
             _autoExtendInterval = TimeSpan.FromSeconds(_modifyDeadlineSeconds) - (settings.AckExtensionWindow ?? DefaultAckExtensionWindow);
+            if (_autoExtendInterval < TimeSpan.FromSeconds(5))
```

Comment:
I'm wondering whether some static methods in GAX would be handy at some point - MaxTimeSpan and MinTimeSpan.